apiserver/facades/client/imagemetadatamanager/metadata.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package imagemetadatamanager
+
+import (
+	"sort"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/common/imagecommon"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/permission"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/cloudimagemetadata"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+var logger = loggo.GetLogger("juju.apiserver.imagemetadatamanager")
+
+// API is the concrete implementation of the api end point
+// for loud image metadata manipulations.
+type API struct {
+	metadata   metadataAccess
+	newEnviron func() (environs.Environ, error)
+	authorizer facade.Authorizer
+}
+
+// createAPI returns a new image metadata API facade.
+func createAPI(
+	st metadataAccess,
+	newEnviron func() (environs.Environ, error),
+	resources facade.Resources,
+	authorizer facade.Authorizer,
+) (*API, error) {
+	if !authorizer.AuthClient() && !authorizer.AuthController() {
+		return nil, common.ErrPerm
+	}
+
+	return &API{
+		metadata:   st,
+		newEnviron: newEnviron,
+		authorizer: authorizer,
+	}, nil
+}
+
+// NewAPI returns a new cloud image metadata API facade.
+func NewAPI(
+	st *state.State,
+	resources facade.Resources,
+	authorizer facade.Authorizer,
+) (*API, error) {
+	newEnviron := func() (environs.Environ, error) {
+		return stateenvirons.GetNewEnvironFunc(environs.New)(st)
+	}
+	return createAPI(getState(st), newEnviron, resources, authorizer)
+}
+
+// List returns all found cloud image metadata that satisfy
+// given filter.
+// Returned list contains metadata ordered by priority.
+func (api *API) List(filter params.ImageMetadataFilter) (params.ListCloudImageMetadataResult, error) {
+	if api.authorizer.AuthClient() {
```

Comment:
the whole facade should only allow clients now, so drop this check, and ones in other non-constructor methods